common/test/acceptance/tests/helpers.py

```diff
@@ -354,6 +354,144 @@ def verify_events_of_type(self, username, event_type, expected_events, expected_
                     "Refer '{0}' does not have correct suffix, '{1}'.".format(actual_referer, expected_referers[index])
                 )
 
+    def create_event_promise(self, check_event_func, description, **kwargs):
+        """
+        Return a promise that is fulfilled when an event results in `check_event_func(event)` returning True.
+
+        Note that `check_event_func` will be called on all events emitted after this moment in time.
+        """
+        return EventPromise(
+            self.event_collection,
+            self.start_time,
+            check_event_func,
+            description,
+            **kwargs
+        )
+
+    def create_event_of_type_promise(self, event_type):
+        """
+        Return a promise that is fulfilled when the first event of the requested type is emitted.
+
+        Note that this does not de-duplicate events, so code like this will not do what you expect::
+
+            promise_a = self.create_event_of_type_promise('foobar')
+            promise_b = self.create_event_of_type_promise('foobar')
+            event_a = promise_a.fulfill()
+            event_b = promise_b.fulfill()
+
+            assert event_a == event_b  # this is true! both of these promises return the first matching event!
+
+        """
+        def check_event_type(event):
+            return event.get('event_type', '') == event_type
+
+        return self.create_event_promise(
+            check_event_type,
+            'Waiting for a {0} event to be emitted'.format(event_type),
+        )
+
+    def assert_event_matches(self, expected, actual, strict=False):
```

Comment:
Right.  I don't know enough about what location could be pulled in from both test frameworks.  But one observation is that the compare_trees (and _path_to_string with it) do not actually need 'self'.  It could be moved anywhere as a standalone method.  The remaining assert that needs self.fail would then be pretty short (code-wise).